driver/i2c: STM32: I2C performance issue solved.

affrinpinhero-2356 opened the pull request:
### Summary of changes <!-- Required -->
This commit solves issue related to i2c driver performance.
With this commit delay in read write when using i2c timing
algorithm is solved. Used flag mechanism which will check
tim reg value and hz passed.
Depends on ARMmbed/mbed-os#14776
since not merged.

#### Impact of changes <!-- Optional -->
Delay in I2C read write while using i2c timing algorithm is solved.


#### Migration actions required <!-- Optional -->

### Documentation <!-- Required -->


----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

[x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
[] Feature update (New feature / Functionality change / New API)
[] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

[] No Tests required for this change (E.g docs only update)
[x] Covered by existing mbed-os tests (Greentea or Unittest)
[] Tests / results supplied as part of this PR

    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->

<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

LMESTM commented on targets/TARGET_STM/i2c_api.c:
```diff
@@ -1728,6 +1754,168 @@ uint32_t i2c_compute_timing(uint32_t clock_src_freq, uint32_t i2c_freq)
+    pclk = i2c_get_pclk(i2c);
+
+    if ((current_timing == 0) || (current_hz != hz)) {
+#if defined (I2C_PCLK_DEF)
```
While rebasing, would you please :
- replace I2C_PCLK_DEF by I2C_PCLK_XXM and TIMING_VAL_DEFAULT_CLK_100KHZ by TIMING_VAL_64M_CLK_100KHZ
- also store the switch case values in from lowest to highest for enhanced readibility  
Thanks

affrinpinhero-2356 committed:
driver/i2c: STM32: Solves I2C driver performanc issue.

This commit solves issue related to i2c driver performance.
With this commit delay in read write when using i2c timing
algorithm is solved. Used flag mechanism which will check
tim reg value and hz passed.

Signed-off-by: Affrin Pinhero <affrin.pinhero@hcl.com>

jeromecoutant reviewed (changes requested):
Non regression tests are failing with F303ZE and L476RG

affrinpinhero-2356 committed:
I2C: STM32: Upating documentation for I2C timing Algorithm

This commit modifies readme file. Descption for using
I2C timing algorithm and how to enable and disable
included.

Signed-off-by: Affrin Pinhero <affrin.pinhero@hcl.com>

0xc0170 commented on targets/TARGET_STM/README.md:
```diff
@@ -408,6 +408,30 @@ You can change this in you local mbed_app.json:
 
+#### I2C TIming calculation algorothm
+
+I2C version 2 drivers are works with help of I2C timing register.
```
I've just noticed. Is this version 2 not just an update to add also timing register for some reason?

```suggestion
I2C drivers version 2 use I2C timing register.
```

0xc0170 commented on targets/TARGET_STM/README.md:
```diff
@@ -408,6 +408,30 @@ You can change this in you local mbed_app.json:
+
+I2C version 2 drivers are works with help of I2C timing register.
+
+In target.json file, has provision to enable and disable I2C timing Algorithm.
```
```suggestion
Enable I2C timing algorithm by setting the value of `i2c_timing_value_algo` target config to `true`.
```

0xc0170 reviewed (dismissed):
Just readme update to fix grammar. sorry for late request, I've noticed it just now
